header/store/options.go

```diff
@@ -0,0 +1,67 @@
+package store
+
+import "errors"
+
+// Option is the functional option that is applied to the store instance
+// to configure store parameters.
+type Option func(*store)
```

Comment:
Pls make the option apply to the Parameters on not the store

Reply:
ooops. 👌